lib/main.dart

```diff
@@ -25,33 +32,55 @@ void main() async {
   environment = configuration.getEnvironment();
   SimpleLogger().info("Running in $environment environment");
 
-  SimpleLogger().info("Connecting to MQTT broker...");
+  deviceId = await keysDao.getDeviceId();
 
-  await _initializeMqttClient();
+  if (deviceId != null) {
+    SimpleLogger().info("Device Id is: $deviceId");
+    SimpleLogger().info("Connecting to MQTT broker...");
+    await mqttClient.connect(deviceId!);
```

Comment:
Does dart still need this assertion to be here even though its within the conditional?